performance_regression_test.py

```diff
@@ -797,3 +798,115 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
 
         self.update_hdrhistograms(histogram_name='test_histogram_by_interval',
                                   histogram_data=histogram_data_by_interval)
+
+
+class PerformanceRegressionUpgradeTest(PerformanceRegressionTest, UpgradeTest):  # pylint: disable=too-many-ancestors
+    def __int__(self, *args):
+        super().__init__(*args)
+        self.cluster = None  # pylint: disable=attribute-defined-outside-init
+        self.tester = None  # pylint: disable=attribute-defined-outside-init
+        self.monitoring_set = None  # pylint: disable=attribute-defined-outside-init
+
+    def get_email_data(self):  # pylint: disable=no-self-use
+        """prepare data to generate and send via email
+
+        Have to return the dict which is used to build the
+        html content with email template.
+        Required field:
+        - email_template: path to file with html template
+        - email_subject: subject of email
+        have to be implemented in child class.
+        """
+        return {}
+
+    @latency_calculator_decorator(legend="Upgrade Node")
+    def _upgrade_node(self, node):
+        InfoEvent(message='Upgrade Node %s begin' % node.name).publish()
+        self.upgrade_node(node)
+        InfoEvent(message='Upgrade Node %s ended' % node.name).publish()
+        node.check_node_health()
+
+    def _stop_stress_when_finished(self):  # pylint: disable=no-self-use
+        with EventsSeverityChangerFilter(new_severity=Severity.NORMAL,  # killing stress creates Critical error
+                                         event_class=CassandraStressEvent,
+                                         extra_time_to_expiration=60):
+            self.loaders.kill_stress_thread()
+
+    @latency_calculator_decorator
+    def steady_state_latency(self):  # pylint: disable=no-self-use
+        sleep_time = 1800
```

Comment:
What is this sleep time?
Why it's 1800 and why it's not controlled from outside parameter?

Reply:
this is THE steady state
this is 1800, because we wanted to reduce the default one (default, based on nemesis interval we have for the config file) that is 3600

if changing this one to the nemesis interval back again, it will be twice as the current value (but controllable from test config file)... so, should i also reduce the `nemesis_interval` (it will affect the nemesis during operations too)